io-engine/src/core/block_device.rs

```diff
@@ -62,6 +62,9 @@ pub trait BlockDevice {
     /// Checks whether target I/O type is supported by the device.
     fn io_type_supported(&self, io_type: IoType) -> bool;
 
+    /// Checks whether target I/O type is supported by the device.
+    fn io_type_supported_by_device(&self, io_type: IoType) -> bool;
```

Comment:
How is this different from io_type_supported()? Please describe the difference in the comment.